testsuite/cluster-test/src/main.rs

```diff
@@ -144,41 +146,73 @@ fn setup_log() {
 
 struct ClusterUtil {
     cluster: Cluster,
-    aws: Aws,
-    prometheus: Prometheus,
+    aws: Option<Aws>,
+    prometheus: Option<Prometheus>,
 }
 
 struct ClusterTestRunner {
     logs: LogTail,
     cluster: Cluster,
     health_check_runner: HealthCheckRunner,
-    deployment_manager: DeploymentManager,
+    deployment_manager: Option<DeploymentManager>,
     experiment_interval: Duration,
     slack: Option<SlackClient>,
     thread_pool: threadpool::ThreadPool,
 }
 
+fn parse_host_port(s: &str) -> failure::Result<(String, u32)> {
+    let v = s.split(':').collect::<Vec<&str>>();
+    if v.len() != 2 {
+        return Err(format_err!("Failed to parse {:?} in host:port format", s));
+    }
+    let host = v[0].to_string();
+    let port = v[1].parse::<u32>()?;
+    Ok((host, port))
+}
+
 impl ClusterUtil {
     pub fn setup(args: &Args) -> Self {
+        let mut is_host_port_format = true;
+        let mut parsed_peers = vec![];
+        for peer in &args.peers {
+            if let Ok(res) = parse_host_port(peer) {
+                parsed_peers.push(res);
+            } else {
+                is_host_port_format = false;
+                break;
+            }
+        }
+        if !parsed_peers.is_empty() && is_host_port_format {
```

Comment:
How about [this](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect)?
```
let res: Result<Vec<_>> = args.peers.iter().map(|peer| parse_host_port(peer)).collect();

if let OK(parsed_peers) = res {
```
?

Reply:
Much better! Updated.